radix-engine/src/system/system.rs

```diff
@@ -2869,99 +2884,7 @@ where
     feature = "std",
     catch_unwind(crate::utils::catch_unwind_system_panic_transformer)
 )]
-impl<'a, Y, V> ClientCryptoUtilsApi<RuntimeError> for SystemService<'a, Y, V>
-where
-    Y: KernelApi<System<V>>,
-    V: SystemCallbackObject,
-{
-    #[trace_resources(log=message.len())]
-    fn bls12381_v1_verify(
-        &mut self,
-        message: &[u8],
-        public_key: &Bls12381G1PublicKey,
-        signature: &Bls12381G2Signature,
-    ) -> Result<u32, RuntimeError> {
-        self.api.kernel_get_system().modules.apply_execution_cost(
-            ExecutionCostingEntry::Bls12381V1Verify {
-                size: message.len(),
-            },
-        )?;
-        Ok(verify_bls12381_v1(message, public_key, signature) as u32)
-    }
-
-    // Trace average message length and number of public_keys
-    #[trace_resources(log={pub_keys_and_msgs.iter().flat_map(|(_, msg)| msg).count()/pub_keys_and_msgs.len()},log=pub_keys_and_msgs.len())]
-    fn bls12381_v1_aggregate_verify(
-        &mut self,
-        pub_keys_and_msgs: &[(Bls12381G1PublicKey, Vec<u8>)],
-        signature: &Bls12381G2Signature,
-    ) -> Result<u32, RuntimeError> {
-        if !pub_keys_and_msgs.is_empty() {
-            let sizes: Vec<usize> = pub_keys_and_msgs.iter().map(|(_, msg)| msg.len()).collect();
-            self.api.kernel_get_system().modules.apply_execution_cost(
-                ExecutionCostingEntry::Bls12381V1AggregateVerify {
-                    sizes: sizes.as_slice(),
-                },
-            )?;
-            Ok(aggregate_verify_bls12381_v1(pub_keys_and_msgs, signature) as u32)
-        } else {
-            Err(RuntimeError::SystemError(SystemError::InputDataEmpty))
-        }
-    }
-
-    #[trace_resources(log=message.len(), log=public_keys.len())]
-    fn bls12381_v1_fast_aggregate_verify(
-        &mut self,
-        message: &[u8],
-        public_keys: &[Bls12381G1PublicKey],
-        signature: &Bls12381G2Signature,
-    ) -> Result<u32, RuntimeError> {
-        if !public_keys.is_empty() {
-            self.api.kernel_get_system().modules.apply_execution_cost(
-                ExecutionCostingEntry::Bls12381V1FastAggregateVerify {
-                    size: message.len(),
-                    keys_cnt: public_keys.len(),
-                },
-            )?;
-            Ok(fast_aggregate_verify_bls12381_v1(message, public_keys, signature) as u32)
-        } else {
-            Err(RuntimeError::SystemError(SystemError::InputDataEmpty))
-        }
-    }
-
-    #[trace_resources(log=signatures.len())]
```

Comment:
`trace_resource` macro should be moved to `Bls12381G2Signature::aggregate()` function.